### PR TITLE
Fix FCOMPmem and FUCOMPmem

### LINKED ISSUE #504: Infinite recursion in the semantics
It seems that the implementation of the FCOMP (and FUCOMP) with memory argument results in an infinite recursion:

```
template <typename S2>
DEF_FPU_SEM(FUCOMPmem, RF80 src1, S2 src2) {
  SetFPUDp(src2);
  return FUCOMPmem(memory, state, src1, src2, pc, fop);
}

template <typename S2>
DEF_FPU_SEM(FCOMPmem, RF80 src1, S2 src2) {
  SetFPUDp(src2);
  return FCOMPmem(memory, state, src1, src2, pc, fop);
}
```

Probably some other function should have been called...

### PR DESCRIPTION
**Pull request by DCNick3**
Closes #504

### CODE REVIEW

File: lib/Arch/X86/Semantics/X87.cpp [hunk 1]
```diff
@@ -1112,13 +1112,17 @@ DEF_FPU_SEM(FCOMP, RF80 src1, S2 src2) {
 template <typename S2>
 DEF_FPU_SEM(FUCOMPmem, RF80 src1, S2 src2) {
   SetFPUDp(src2);
-  return FUCOMPmem(memory, state, src1, src2, pc, fop);
+  memory = FUCOMP(memory, state, src1, src2, pc, fop);
+  (void) POP_X87_STACK();
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/X87.cpp:1116]: Should a double pop be happening? `FUCOMP` pops already, I believe.

File: lib/Arch/X86/Semantics/X87.cpp [hunk 2]
```diff
@@ -1112,13 +1112,17 @@ DEF_FPU_SEM(FCOMP, RF80 src1, S2 src2) {
...
+  return memory;
 }
 
 template <typename S2>
 DEF_FPU_SEM(FCOMPmem, RF80 src1, S2 src2) {
   SetFPUDp(src2);
-  return FCOMPmem(memory, state, src1, src2, pc, fop);
+  memory = FCOMP(memory, state, src1, src2, pc, fop);
+  (void) POP_X87_STACK();
```
Comment (pgoodman) [inline lib/Arch/X86/Semantics/X87.cpp:1124]: Should a double pop be happening? `FCOMP` pops already, I believe.

File: lib/Arch/X86/Semantics/X87.cpp (hunk 1 above)
Comment (DCNick3) [reply lib/Arch/X86/Semantics/X87.cpp]: Yeah, you are right

File: lib/Arch/X86/Semantics/X87.cpp (hunk 2 above)
Comment (DCNick3) [reply lib/Arch/X86/Semantics/X87.cpp]: Yeah, you are right

Comment (DCNick3) [reply lib/Arch/X86/Semantics/X87.cpp]: Yeah, you are right

**Comment by DCNick3** [comment]
Does not seem to me that this CI failure is related to my changes...

Intel manual says this about the mismatching leaf:
image

Seems to me this is environment-specific, probably should be ignored in the test case

**Review by pgoodman — APPROVED** [review]

**Pull request merged**